docs/content/policy-reference.md

```diff
@@ -1111,28 +1111,31 @@ import          = "import" ref [ "as" var ]
 policy          = { rule }
 rule            = [ "default" ] rule-head { rule-body }
 rule-head       = var ( rule-head-set | rule-head-obj | rule-head-func | rule-head-comp )
-rule-head-comp  = [ ( ":=" | "=" ) term ]) [ "if" ]
-rule-head-obj   = [ "[" term "]" ] [ ( ":=" | "=" ) term ]) [ "if" ]
-rule-head-func  = [ "(" rule-args ")" ] [ ( ":=" | "=" ) term ]) [ "if" ]
+rule-head-comp  = [ assign-operator term ] [ "if" ]
+rule-head-obj   = "[" term "]" [ assign-operator term ] [ "if" ]
+rule-head-func  = "(" rule-args ")" [ assign-operator term ] [ "if" ]
 rule-head-set   = "contains" term [ "if" ] | "[" term "]"
 rule-args       = term { "," term }
-rule-body       = [ "else" [ ( ":=" | "=" ) term ] ] "{" query "}"
+rule-body       = [ "else" [ assign-operator term ] ] "{" query "}"
 query           = literal { ( ";" | ( [CR] LF ) ) literal }
 literal         = ( some-decl | expr | "not" expr ) { with-modifier }
 with-modifier   = "with" term "as" term
 some-decl       = "some" term { "," term } { "in" expr }
-expr            = term | expr-call | expr-infix | expr-every
+expr            = term | expr-call | expr-infix | expr-every | expr-parens
 expr-call       = var [ "." var ] "(" [ expr { "," expr } ] ")"
-expr-infix      = [ term "=" ] expr infix-operator expr
+expr-infix      = expr infix-operator expr
 expr-every      = "every" var { "," var } "in" ( term | expr-call | expr-infix ) "{" query "}"
-term            = ref | var | scalar | array | object | set | array-compr | object-compr | set-compr
-array-compr     = "[" term "|" rule-body "]"
-set-compr       = "{" term "|" rule-body "}"
-object-compr    = "{" object-item "|" rule-body "}"
-infix-operator  = bool-operator | arith-operator | bin-operator
+expr-parens     = "(" expr ")"
+membership      = scalar [ "," scalar ] "in" term
```

Comment:
I don't think `scalar` is correct here, but `term` might be. For example,
```rego
package play

import future.keywords.in
import future.keywords.if

hello if input.x, input.y in input.z
```

Reply:
@srenatus Ah, that makes a lot of sense! I'll update the grammar to:

```ebnf
membership = term [ "," term ] "in" term
```

However, looking at your example, I think we have some more work to do :) IIUIC the example should be parsed roughly as:

```rego
hello if input.x, input.y in input.z
^^^^^ rule-head

      ^^ "if"

         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ rule-body
```

So if my understanding is correct, we're also going to need to update `rule-body` to match against `literal` directly as it currently matches only `query` enclosed in `{}`:

```diff
-rule-body       = [ "else" [ ( ":=" | "=" ) term ] ] "{" query "}"
+rule-body       = [ "else" [ ( ":=" | "=" ) term ] ] ( "{" query "}" ) | literal
```

I'll soon update the playground with all of these changes to see how it plays out.

Reply:
Yes that sounds about right. Thanks for taking care of those neglected bits, I really appreciate it!

Reply:
It's a little tricky since what you've mentioned is only valid if the proper future import is there, but that's not something to capture by the means of a grammar, I suppose.

Reply:
I think this bit kinda saves our backs :)

```
Note that the grammar corresponds to Rego with all future keywords enabled.
```